CMakeLists.txt

```diff
@@ -1,67 +1,46 @@
 cmake_minimum_required(VERSION 3.12)
 project(slingsim)
 
-# Compilation flags definition
-set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_CXX_FLAGS_DEBUG "-Wall -Wextra")
+# Main target creation
+add_executable(${PROJECT_NAME})
 
-find_program(CCACHE_FOUND ccache)
-if(CCACHE_FOUND)
-  message("## Compile with ccache")
-  set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
-endif(CCACHE_FOUND)
+include(tools/cmake/Libs.cmake)
 
-# Operating system configuration
-option(UNIX_COMPILATION "Compile For Unix systems" ON)
-if(UNIX_COMPILATION)
+# Operating system specific settings
+set(CMAKE_CXX_STANDARD 20)
+if(UNIX)
   message("## Compile for unix operating system")
-  set(CMAKE_BINARY_DIR ${CMAKE_HOME_DIRECTORY}/build/bin)
-else()
-  # Set the compile flags to statically link the  libgcc dll files, this makes a standalone exe
-  message("## Compile for non unix operating system")
-  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_DEBUG} -static-libgcc -static-libstdc++")
-  set(CMAKE_BINARY_DIR ${CMAKE_HOME_DIRECTORY}/build/bin)
-endif()
-
-# Advanced compilation options
-option(ENABLE_TESTS "Compile and run unit tests" ON)
-option(ENABLE_DOCUMENTATION "Build documentation target" ON)
-option(ENABLE_COVERAGE "Build coverage target" ON)
-option(ENABLE_DIAGNOSTIC_TOOLS "Build with diagnostic tools" OFF)
-
```

Comment:
I'm gonna need these for unix
# Advanced compilation options
option(ENABLE_TESTS "Compile and run unit tests" ON)
option(ENABLE_DOCUMENTATION "Build documentation target" ON)
option(ENABLE_COVERAGE "Build coverage target" ON)
option(ENABLE_DIAGNOSTIC_TOOLS "Build with diagnostic tools" OFF)

These options are platform agnostic and tell the cmake to add various types of compilation options.
If you define inside the included source it does not get read correctly.

Reply:
You are right, I already fixed it